src/core/cm/imagemanager/imagemanager.cpp

```diff
@@ -952,4 +1065,402 @@ Error ImageManager::CleanupOrphanedBlobs(
     return ErrorEnum::eNone;
 }
 
+Error ImageManager::RemoveDifferentVersions(const Array<UpdateItemInfo>& itemsInfo, const Array<ItemInfo>& storedItems)
+{
+    LOG_DBG() << "Remove different versions";
+
+    for (const auto& storedItem : storedItems) {
+        auto requestedIt
+            = itemsInfo.FindIf([&storedItem](const auto& item) { return item.mItemID == storedItem.mItemID; });
+
+        if (requestedIt != itemsInfo.end()) {
+            if (requestedIt->mVersion != storedItem.mVersion) {
+                LOG_DBG() << "Wiping different version" << Log::Field("itemID", storedItem.mItemID)
+                          << Log::Field("storedVersion", storedItem.mVersion)
+                          << Log::Field("requestedVersion", requestedIt->mVersion);
+
+                if (auto err = mStorage->RemoveItem(storedItem.mItemID, storedItem.mVersion); !err.IsNone()) {
+                    LOG_ERR() << "Failed to remove item from storage" << Log::Field("itemID", storedItem.mItemID)
+                              << Log::Field("version", storedItem.mVersion) << Log::Field(err);
+                }
+
+                if (!storedItem.mIndexDigest.IsEmpty()) {
+                    auto filePath = fs::JoinPath(mBlobsInstallPath, storedItem.mIndexDigest);
+                    LOG_DBG() << "Wipe blob" << Log::Field("path", filePath);
+
+                    if (auto err = fs::RemoveAll(filePath); !err.IsNone()) {
```

Comment:
Unclear why only index file is removed? What about manifests?

IMHO here we should remove only DB entry or all related blobs.

Reply:
removed. Not index blob removed in `CleanupDownloadingItems`